File: 2nd Ed week 13/leetcode/lucky-numbers-in-a-matrix.py
class Solution:
    def luckyNumbers (self, matrix: List[List[int]]) -> List[int]:
        n = len(matrix)
        m = len(matrix[0])

        minimums = []
        for i in range(n):
            minimums.append(min(matrix[i]))

        maximums = []
        for j in range(m):
            cur_max = matrix[0][j]
            for i in range(1, n):
                cur_max = max(cur_max, matrix[i][j])
            
            maximums.append(cur_max)
        
        answer = []
        for i in range(n):
            for j in range(m):
                if matrix[i][j] == minimums[i] and matrix[i][j] == maximums[j]:
                    answer.append(matrix[i][j])
        
        return answer

File: 2nd Ed week 13/leetcode/lucky-numbers-in-a-matrix.go
func luckyNumbers (matrix [][]int) []int {
    n := len(matrix)
    m := len(matrix[0])

    var minimums []int
    var maximums []int

    for i:=0; i < n; i++ {
        cur_min := matrix[i][0]
        for j := 1; j < m; j++{
            if cur_min >= matrix[i][j]{
                cur_min = matrix[i][j]
            }
        }
        minimums = append(minimums, cur_min)
    }
    
    for j := 0; j < m; j++{
        cur_max := matrix[0][j]
        for i := 1; i < n; i++{
            if cur_max < matrix[i][j]{
                cur_max = matrix[i][j]
            }
        }
        maximums = append(maximums, cur_max)
    }

    var answer []int
    
    for i := 0; i < n; i++{
        for j := 0; j < m; j++{
            if matrix[i][j] == minimums[i] && matrix[i][j] == maximums[j]{
                answer = append(answer, matrix[i][j])
            }
        }
    }

    return answer
}